tracer/src/Datadog.Trace/OTelMetrics/TagSet.cs

```diff
@@ -25,16 +26,55 @@ namespace Datadog.Trace.OTelMetrics;
 
     public static TagSet FromSpan(ReadOnlySpan<KeyValuePair<string, object?>> tags)
     {
-        if (tags.Length == 0)
+        var len = tags.Length;
+        if (len == 0)
         {
             return new TagSet(string.Empty);
         }
 
-        var sorted = tags.ToArray();
-        Array.Sort(sorted, (a, b) => string.CompareOrdinal(a.Key, b.Key));
+        if (len == 1)
+        {
+            ref readonly var kv = ref tags[0];
+            var sb = StringBuilderCache.Acquire(32 + kv.Key.Length);
+            sb.Append(kv.Key).Append('=');
+            if (kv.Value is not null)
+            {
+                sb.Append(kv.Value);
+            }
+
+            return new TagSet(StringBuilderCache.GetStringAndRelease(sb));
+        }
+
+        var sorted = ArrayPool<KeyValuePair<string, object?>>.Shared.Rent(len);
+        try
+        {
+            tags.CopyTo(sorted);
+            Array.Sort(sorted, 0, len, Comparer<KeyValuePair<string, object?>>.Create(
```

Comment:
I think you should create the Comparer once (the third argument to the `Array.Sort` call) and store it in a `static readonly Comparer<KeyValuePair<string, object?>>` field, so you only invoke that API once, rather than every time we need to construct a TagSet